core/request.py

```diff
@@ -527,41 +534,42 @@ def findAndCall(self, path: str, *args, **kwargs) -> None:
         caller = conf["viur.mainResolver"]
         idx = 0  # Count how may items from *args we'd have consumed (so the rest can go into *args of the called func
-        for currpath in self.pathlist:
+        path_found = True
+        for part in self.path_list:
             if "canAccess" in caller and not caller["canAccess"]():
                 # We have a canAccess function guarding that object,
                 # and it returns False...
                 raise (errors.Unauthorized())
             idx += 1
-            currpath = currpath.replace("-", "_").replace(".", "_")
-            if currpath in caller:
-                caller = caller[currpath]
-                if (("exposed" in dir(caller) and caller.exposed) or ("internalExposed" in dir(
-                    caller) and caller.internalExposed and self.internalRequest)) and hasattr(caller, '__call__'):
-                    args = self.pathlist[idx:] + [x for x in args]  # Prepend the rest of Path to args
+            part = part.replace("-", "_").replace(".", "_")
+            if part not in caller:
+                part = "index"
+
+            if part in caller:
+                caller = caller[part]
+                if (("exposed" in dir(caller) and caller.exposed) or
+                        ("internalExposed" in dir(caller) and caller.internalExposed and self.internalRequest)) and\
+                        hasattr(caller, '__call__'):
+                    if part == "index":
+                        idx -= 1
+                    args = self.path_list[idx:] + args  # Prepend the rest of Path to args
                     break
-            elif "index" in caller:
-                caller = caller["index"]
-                if (("exposed" in dir(caller) and caller.exposed) or ("internalExposed" in dir(
-                    caller) and caller.internalExposed and self.internalRequest)) and hasattr(caller, '__call__'):
-                    args = self.pathlist[idx - 1:] + [x for x in args]
+
+                elif part == "index":
+                    path_found = False
                     break
-                else:
-                    raise (errors.NotFound("The path %s could not be found" % "/".join(
-                        [("".join([y for y in x if y.lower() in "0123456789abcdefghijklmnopqrstuvwxyz"])) for x in
-                         self.pathlist[: idx]])))
+
             else:
-                raise (errors.NotFound("The path %s could not be found" % "/".join(
-                    [("".join([y for y in x if y.lower() in "0123456789abcdefghijklmnopqrstuvwxyz"])) for x in
-                     self.pathlist[: idx]])))
+                path_found = False
+                break
+
+        if not path_found:
+            raise errors.NotFound("The path %s could not be found" % "/".join(
+                    [("".join([char for char in part if char.lower() in string.ascii_lowercase+string.digits]))
+                     for part in self.path_list[: idx]]))
```

Comment:
I wonder what's the intense behind the character-ASCII-DIGIT-check stuff here. Security? Then, why isn't this already done in line 500? 

IMHO, this can be cut down to this:
```suggestion
            raise errors.NotFound(f"""The path {"/".join(self.path_list[:idx]!a} could not be found""")
```